lib/cassandra/timestamp_generator/ticking_on_duplicate_generator.rb

```diff
@@ -0,0 +1,58 @@
+# encoding: utf-8
+
+#--
+# Copyright 2013-2016 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+module Cassandra
+  module TimestampGenerator
+    # In JRuby, {::Time} has millisecond precision. We require client timestamps to have microsecond precision to
+    # minimize clashes in C*. This generator keeps track of the last generated timestamp, and if the current-time
+    # is within the same millisecond as the last, it fills the microsecond portion of the new timestamp with the
+    # value of an incrementing counter.
+    #
+    # For example, if the generator triggers twice at time 12345678000 (microsecond granularity, but ms precisions
+    # as shown by 0's for the three least-significant digits), it'll return 12345678000 and 12345678001.
+    class TickingOnDuplicate
```

Comment:
ahh, right. My fingers were going faster than my brain (or the other way around).